### ISSUE TITLE
Broken GEXF Files Due to < and > characters in node id fields

### ISSUE THREAD

**Issue by ianmilligan1** [archivesunleashed/aut#172]
We've been generating lots of GEXF files. Occasionally, we extract very broken links into files like:

```
<node id="<a href=" label="<a href=" />
<node id="w<" label="w<" />
<node id="rstyle....<a href=" label="rstyle....<a href=" />
<node id="calgarycrimestoppers...<" label="calgarycrimestoppers...<" />
<node id="bath.ac.uk%2Face%2Fuploads%2FStudentProjects%2FBridgeconference2009%2FPapers%2FGRIFFITHS.pdf&ei=FFPpUOylI-LG0QWIoIDwAg&usg=AFQjCNGVHMfH3srFSEklt9ixW2vSR8AtIw&bvm=bv.1355534169,d.d2k" label="bath.ac.uk%2Face%2Fuploads%2FStudentProjects%2FBridgeconference2009%2FPapers%2FGRIFFITHS.pdf&ei=FFPpUOylI-LG0QWIoIDwAg&usg=AFQjCNGVHMfH3srFSEklt9ixW2vSR8AtIw&bvm=bv.1355534169,d.d2k" />
<node id="WhatsYourTech.ca"" label="WhatsYourTech.ca"" />
```

Note above some of the errors: `<` ending up in id fields, or two closing `"` in an `id` field, etc. All of this leads to Gephi explosions like so:

```
javax.xml.stream.XMLStreamException: ParseError at [row,col]:[3472,18]
Message: The value of attribute "id" associated with an element type "node" must not contain the '<' character.
	at com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.next(XMLStreamReaderImpl.java:604)
	at org.gephi.io.importer.plugin.file.ImporterGEXF.execute(ImporterGEXF.java:145)
Caused: java.lang.RuntimeException
	at org.gephi.io.importer.plugin.file.ImporterGEXF.execute(ImporterGEXF.java:169)
	at org.gephi.io.importer.impl.ImportControllerImpl.importFile(ImportControllerImpl.java:199)
	at org.gephi.io.importer.impl.ImportControllerImpl.importFile(ImportControllerImpl.java:169)
	at org.gephi.desktop.importer.DesktopImportControllerUI$4.run(DesktopImportControllerUI.java:341)
Caused: java.lang.RuntimeException
	at org.gephi.desktop.importer.DesktopImportControllerUI$4.run(DesktopImportControllerUI.java:349)
[catch] at org.gephi.utils.longtask.api.LongTaskExecutor$RunningLongTask.run(LongTaskExecutor.java:274)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by cjer** [comment]
I can give a few more examples of bad links - will do so tomorrow

**Comment by greebie** [comment]
Scala does have an html escaping utility that we can add to the list of UDFs.  
http://www.scala-lang.org/api/2.8.1/scala/xml/Utility$.html

I just need to figure out if it's better to add it as a UDF or hide it behind the WriteGEXF & WriteGraphML functions. Probably the former for the case where someone wants to do their own *ML-based output.

I thought I'd get to this today, but it looks like it may take until Tuesday.

**Comment by ianmilligan1** [comment]
Closed with #173
